### PR TITLE
Be able to mark javalab files as support or starter in levelbuilder

### PR DESCRIPTION
**Pull request by JillianK**
Adds a field to levelbuilder on javalab levels that displays all of the files in the level and has a dropdown to set a file as Starter (visible) or Support (not visible).

Screen Shot 2021-04-27 at 12 39 38 PM
Screen Shot 2021-04-27 at 12 38 32 PM

## Links

- design in first screenshot here: [CSA Design Level Building](https://docs.google.com/document/d/1m7s_DF2JM4ZRbbihj3HsuAA3n-wePegLvcwFQjEx2sA/edit#heading=h.bj1j0htqouu)
- jira ticket: [CSA-238](https://codedotorg.atlassian.net/browse/CSA-238?atlOrigin=eyJpIjoiZjZjOTlkNDgwMDk4NGRjYmI1YjQwMzFkNmQ5NWRkMjkiLCJwIjoiaiJ9)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/levels_controller.rb [hunk 1]
```diff
@@ -428,6 +428,7 @@ def level_params
       {if_block_options: []},
       {place_block_options: []},
       {play_sound_options: []},
+      {visible: {}},
```
Comment (molly-moen) [inline dashboard/app/controllers/levels_controller.rb:431]: maybe put a comment here that this is CSA-specific? right now it looks like it's in the Minecraft-specific category

File: dashboard/app/models/levels/javalab.rb [hunk 2]
```diff
@@ -52,6 +52,12 @@ def fix_examples
     self.examples = examples.select(&:present?)
   end
 
+  def assign_attributes(params)
+    params[:visible].keys.map(&:to_s).each do |filename|
```
Comment (molly-moen) [inline dashboard/app/models/levels/javalab.rb:56]: I think we may need a check here to see if params[:visible] exists or not

File: dashboard/app/controllers/levels_controller.rb (hunk 1 above)
Comment (jmkulwik) [reply dashboard/app/controllers/levels_controller.rb:431]: Additionally, the variable name makes it appear that we're defining whether the level itself is visible. Suggested renames: `start_sources_metadata`, `files_metadata`, `visible_files`, `file_view_types` or something similar.

File: dashboard/app/models/levels/javalab.rb [hunk 3]
```diff
@@ -52,6 +52,12 @@ def fix_examples
...
+      start_sources[filename]["visible"] = params[:visible][filename] === "true"
```
Comment (jmkulwik) [inline dashboard/app/models/levels/javalab.rb:57]: What should happen if `start_sources[filename]` or `params[:visible][filename]` doesn't exist? (would that ever happen)

**Comment by JillianK** [comment]
We decided to move away from this and put the setting in the editor tab dropdown menu instead. See [this PR](https://github.com/code-dot-org/code-dot-org/pull/40360) for more details

**Pull request closed without merging**